src/client/Client.cc

```diff
@@ -11345,6 +11345,10 @@ int64_t Client::_preadv_pwritev_locked(Fh *fh, const struct iovec *iov,
     if (fh->flags & O_PATH)
         return -EBADF;
 #endif
+    // negative iovcount is invalid,
+    // if we are doing zerocopy, we must also have a bufferlist
+    // zero copy could work with a passed in buffer for write 
+    // (but not for read - for read, we need an iovec to fill, a passed in buffer would force a copy).
```

Comment:
Hmm, though here, the param is a bufferlist... We COULD take a bufferlist that we fill...

Reply:
So maybe the correct check will be

```
iif (iovcnt < 0 || (zerocopy && (!blp || (!write && !iov)))) {
    return -EINVAL;
}
```
And it can be used both in `libcephfs.cc` and `Client.ccc` ?